### PR TITLE
Clarify use of application-default credentials

### LINKED ISSUE #283: Documentation not clear if I should use application default credentials on production
The README says:

> if you are running your application on a machine where your are using the gcloud command line tools, and are logged using gcloud beta auth application-default login, you already have sufficient credentials, and a service account key is not required.

Does this mean that on a production GCE instance, it is OK to use application default credentials generated by gcloud? 
If not, I think we should change the doc to say to only use application default credentials on a local test environment.

### PR DESCRIPTION
**Pull request by matthewloring**
Specify that application default credentials is only recommended for
development/testing environments.

Fixes #283

### CODE REVIEW

**Comment by matthewloring** [comment]
@steren @GoogleCloudPlatform/node-team PTAL.

**Comment by ofrobots** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ If your application is running outside of Google Cloud Platform, such as locally
   2. Copy the key somewhere your application can access it. Be sure not to expose the key publicly.
   3. Set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to the full path to the key. The debug agent will automatically look for this environment variable.
 
-3. Alternatively, if you are running your application on a machine where your are using the [`gcloud` command line tools][gcloud-sdk], and are logged using `gcloud beta auth application-default login`, you already have sufficient credentials, and a service account key is not required.
+3. Alternatively, if you are running your application on a development machine or test environment where your are using the [`gcloud` command line tools][gcloud-sdk], and are logged using `gcloud beta auth application-default login`, you already have sufficient credentials, and a service account key is not required.
```
Comment (kjin) [inline README.md]: Small change, but since it's on the same line - change your -> you

Comment (matthewloring) [inline README.md]: Good catch! Fixed.

**Comment by steren** [comment]
LGMT

**Pull request merged**